crypto/property/property_parse.c

```diff
@@ -117,15 +126,27 @@ static int parse_hex(const char *t[], OSSL_PROPERTY_DEFINITION *res)
 {
     const char *s = *t;
     int64_t v = 0;
+    int sval;
 
-    if (!ossl_isxdigit(*s))
-        return 0;
     do {
-        v <<= 4;
         if (ossl_isdigit(*s))
-            v += *s - '0';
-        else
-            v += ossl_tolower(*s) - 'a';
+            sval = *s - '0';
+        else if (ossl_isxdigit(*s))
+            sval = ossl_tolower(*s) - 'W';
```

Comment:
That really isn't clear code - it should be subtract 'a' **and add 10** logically.
This has nothing to do with 'W' and with no comment about it that really doesn't work well.

Note this also won't work on EBCDIC systems ...

Reply:
ascii 'W' is the equivalent of subtracting ascii 'a' and adding 10, but if we support EBCDIC encoding I can make the appropriate change

Reply:
I know what the code is doing - but writing "W" *obfuscates* the logic - so both the obfuscation and the non-EBCDIC issue apply - i.e. even if you ignore EBCDIC that code should be changed IMHO.